From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Keishi Hattori <keishi@chromium.org>
Date: Wed, 5 Apr 2023 04:03:14 +0000
Subject: Revert "Reland "[BRP] Enable clang plugin check for Linux""

This reverts commit f4d31c1eea33e9d295a6c9ca6b9110c06adf9c05.

Reason for revert: Broke ANGLE roll.
https://crbug.com/1430536

Original change's description:
> Reland "[BRP] Enable clang plugin check for Linux"
>
> This is a reland of commit cb1754cef1a02f6a6270389acf4a67d4bfea03a5
> The build error on ASan Release (32-bit x86 with V8-ARM)
> was fixed with this CL: https://crrev.com/c/4393744
>
> Original change's description:
> > [BRP] Enable clang plugin check for Linux
> >
> > clang will now show warnings where raw_ptr<T> could be used.
> >
> > Bug: 1273182
> > Change-Id: I524774dbcf892180961151bc9dc7d8c1c711d383
> > Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4379698
> > Reviewed-by: Takuto Ikuta <tikuta@chromium.org>
> > Reviewed-by: Bartek Nowierski <bartekn@chromium.org>
> > Commit-Queue: Keishi Hattori <keishi@chromium.org>
> > Cr-Commit-Position: refs/heads/main@{#1124604}
>
> Bug: 1273182
> Change-Id: I23ab2a733ccec86d865258bc7d128d01971aa26f
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4393907
> Reviewed-by: Bartek Nowierski <bartekn@chromium.org>
> Reviewed-by: Takuto Ikuta <tikuta@chromium.org>
> Commit-Queue: Keishi Hattori <keishi@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1125241}

Bug: 1273182
Change-Id: Id6c45aa560dac4583f466dadbea117fe88cdca79
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4401993
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Reviewed-by: Bartek Nowierski <bartekn@chromium.org>
Commit-Queue: Keishi Hattori <keishi@chromium.org>
Reviewed-by: Takuto Ikuta <tikuta@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1126401}
(cherry picked from commit 827ce36aa5e210b3b80785a92d880ea225f3c95a)

diff --git a/build/config/clang/BUILD.gn b/build/config/clang/BUILD.gn
index ab99fbedc808368c6e058268b69dcde2b6826371..fbe13243937208ade6ee0388007b9bebaaf7eba0 100644
--- a/build/config/clang/BUILD.gn
+++ b/build/config/clang/BUILD.gn
@@ -35,27 +35,6 @@ config("find_bad_constructs") {
         "check-ipc",
       ]
     }
-
-    if (!is_official_build && is_linux && !is_castos) {
-      cflags += [
-        "-Xclang",
-        "-plugin-arg-find-bad-constructs",
-        "-Xclang",
-        "check-raw-ptr-fields",
-
-        # TODO(keishi): Remove this once crrev.com/c/4387753 is rolled out.
-        "-Xclang",
-        "-plugin-arg-find-bad-constructs",
-        "-Xclang",
-        "raw-ptr-exclude-path=base/no_destructor.h",
-
-        # TODO(keishi): Remove this once crrev.com/c/4086161 lands.
-        "-Xclang",
-        "-plugin-arg-find-bad-constructs",
-        "-Xclang",
-        "raw-ptr-exclude-path=base/containers/span.h",
-      ]
-    }
   }
 }
 
